### PR TITLE
[logging] add MLR log region

### PR DESCRIPTION
**Pull request by simonlingoogle**
Adds the `MLR` log region, which is going to be used by MLR features.

**MLR PR List:**
- #5248 Send MLR.req for locally Netfi MAs 
- #5236 Parent send MLR.req on behave of Children 
- #5250 Add the `MLR` log region  **<== This PR**

### CODE REVIEW

File: src/core/common/logging.hpp [hunk 1]
```diff
@@ -483,6 +485,64 @@ extern "C" {
...
+ * @def otLogWarnMlr
+ *
+ * This method generates a log with level warning for the Multicast Listener Registration (MLR) region.
+ *
+ * @param[in]  ...  Arguments for the format specification.
+ *
+ */
+
+/**
+ * @def otLogInfoMlr
```
Comment (Irving-cl) [inline src/core/common/logging.hpp:507]:
```suggestion
 * @def otLogNoteMlr
```

Comment (simonlingoogle) [reply src/core/common/logging.hpp:507]: I copied from `otLogInfoBbr`. So there should be issue there too.

File: src/ncp/ncp_base.cpp [hunk 2]
```diff
@@ -612,6 +612,9 @@ unsigned int NcpBase::ConvertLogRegion(otLogRegion aLogRegion)
     case OT_LOG_REGION_BBR:
         spinelLogRegion = SPINEL_NCP_LOG_REGION_OT_BBR;
         break;
+    case OT_LOG_REGION_MLR:
+        spinelLogRegion = SPINEL_NCP_LOG_REGION_OT_BBR;
```
Comment (librasungirl) [inline src/ncp/ncp_base.cpp:616]:
```suggestion
        spinelLogRegion = SPINEL_NCP_LOG_REGION_OT_MLR;
```

File: src/core/common/logging.hpp [hunk 3]
```diff
@@ -483,6 +485,64 @@ extern "C" {
...
+
+/**
+ * @def otLogDebgMlr
+ *
+ * This method generates a log with level debug for the Multicast Listener Registration (MLR) region.
+ *
+ * @param[in]  ...  Arguments for the format specification.
+ *
+ */
+#if OPENTHREAD_CONFIG_LOG_MLR == 1
```
Comment (librasungirl) [inline src/core/common/logging.hpp:532]: add default definition in `src/core/config/logging.h` to suppress the warning?

**Pull request merged**